### PR TITLE
Remove akka dependencies from core module

### PR DESCRIPTION
**Pull request by neerajrj**
### Context
Mantis Jobs inadvertently pull in akka dependencies due to the nfmantis-runtime depending on mantis-control-plane-core (which has these dependencies)
This PR moves these dependencies to the server module

### Checklist

- [ x] `./gradlew build` compiles code correctly
- [ x] Added new tests where applicable
- [ x] `./gradlew test` passes all tests
- [ x] Extended README or added javadocs where applicable
- [ x] Added copyright headers for new files from `CONTRIBUTING.md`

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -33,15 +33,7 @@ ext {
...
-    api "com.typesafe.akka:akka-http-jackson_2.13:$akkaHttpVersion"
-    api "com.typesafe.akka:akka-http-caching_2.13:$akkaHttpVersion"
-    api "com.typesafe.akka:akka-stream_2.13:$akkaVersion"
-    compile group: 'com.typesafe.akka', name: 'akka-stream_2.13', version: '2.5.23'
-    api "com.typesafe.akka:akka-slf4j_2.13:$akkaVersion"
 
-    // Master core
-    api "com.typesafe.akka:akka-actor_2.13:$akkaVersion"
     api "io.mantisrx:mantis-common:$mantisVersion"
     api "io.reactivex:rxjava-reactive-streams:$rxJavaReactiveStreamsVersion"
```
Comment (nickmahilani) [inline core/build.gradle]: can this move too?

File: core/build.gradle [hunk 2]
```diff
@@ -59,7 +51,7 @@ dependencies {
     api "org.apache.curator:curator-recipes:$curatorVersion"
```
Comment (nickmahilani) [inline core/build.gradle:43]: same for curator stuff, probably not required in control-plane-core module

File: core/build.gradle (hunk 1 above)
Comment (nickmahilani) [reply core/build.gradle]: fenzo seems server side as well. lets remove whats not needed by core so we reduce the footprint on mantis-worker

**Review by nickmahilani — CHANGES_REQUESTED** [review]

**Comment by nickmahilani** [comment]
Looks great ! good to not have akka deps proliferating on workers, can probably trim some more deps while we are at it

File: core/build.gradle (hunk 1 above)
Comment (neerajrj) [reply core/build.gradle]: moved fenzo as well

File: core/build.gradle (hunk 2 above)
Comment (neerajrj) [reply core/build.gradle:43]: Will leave curator for now as the worker will need it anyways and removing it here would need to be done in conjunction with adding it in the mantis worker.

File: core/build.gradle (hunk 1 above)
Comment (neerajrj) [reply core/build.gradle]: Move rxjava reactive streams as well

**Review by nickmahilani — APPROVED** [review]

**Pull request merged**